src/services/projectService.ts

```diff
@@ -58,6 +59,29 @@ export default class ProjectService implements IProjectService {
         });
     }
 
+    public isDuplicate(project: IProject, projectList: IProject[]): boolean {
+        const projectId = project.id;
+        const projectName = project.name;
+        const sourceConnection = project.sourceConnection.name;
+        const targetConnection = project.targetConnection.name;
+        if (projectList && projectList.length > 0) {
+            const duplicate = (projectId === undefined) &&
```

Comment:
Won't new projects bypass this check all together now?

Reply:
```
function isDuplicate(project: IProject, projectList: IProject[]): boolean {
    const duplicateProjects = projectList.find((p) => 
      p.id !== project.id && 
      p.name === project.name && 
      JSON.stringify(p.targetConnection.providerOptions) === JSON.stringify(project.targetConnection.providerOptions)
    );

return duplicationProjects.length > 0;
}
```

Reply:
Try something like this.

1. Same name, same id, same target connection => pass (edit scenario)
2. Same name, different id same target connection => fail (dup detected, edit project)
3. Same name, no id, same target conneciton => fail (dup detected, new project)
4. Same name, different id, different target connection => pass (no dup editing project)

Reply:
How does case 2 happen? If this is a blocker from letting them create a duplicate project? How would a project come into existence that has a difference id with the same name and target connection?

Reply:
@elizabethhalper - through an error scenario.. you originally saved it and it was fine... then on edit you try to use a name that already exists with the same target connection.  That should trigger the failure event.

Reply:
@wbreza  The fix I made is close to the code above and should address both scenarios (2 and 3)